step.go

```diff
@@ -400,325 +363,47 @@ func exportOutput(artifacts []string) (string, string, error) {
-	}
-
-	targets := append([]xcodeproj.Target{mainTarget}, proj.DependentTargetsOfTarget(mainTarget)...)
-	for _, target := range targets {
-		log.Donef(target.Name + "...")
-
-		// Is the target an application? -> If not skip the export
-		if !target.IsAppProduct() {
-			log.Printf("Target (%s) is not an .app - SKIP", target.Name)
-			continue
-		}
-
-		//
-		// Find out the sdk for the target
-		{
-			settings, err := proj.TargetBuildSettings(target.Name, configuration)
-			if err != nil {
-				log.Debugf("Failed to fetch project settings (%s), error: %s", proj.Path, err)
-			}
-
-			sdkRoot, err := settings.String("SDKROOT")
-			if err != nil {
-				log.Debugf("No SDKROOT config found for (%s) target", target.Name)
-			}
 
-			log.Debugf("sdkRoot: %s", sdkRoot)
-
-			if strings.Contains(sdkRoot, "WatchOS.platform") {
-				simulatorSDK = watchOSSimSDK
-			}
+	if err := filepath.WalkDir(archivePath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
 		}
 
-		//
-		// Find the TARGET_BUILD_DIR for the target
-		options := []string{"-sdk", string(simulatorSDK)}
-		var targetDir string
-		{
-			if sliceutil.IsStringInSlice("-sdk", customOptions) {
-				options = customOptions
-			} else {
-				options = append(options, customOptions...)
-			}
+		if d.IsDir() && filepath.Ext(d.Name()) == ".app" {
+			destination := filepath.Join(deployDir, d.Name())
 
-			buildSettings, err := proj.TargetBuildSettings(target.Name, configuration, options...)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get project build settings, error: %s", err)
+			cmd := util.CopyDir(path, destination)
+			cmd.SetStdout(os.Stdout)
+			cmd.SetStderr(os.Stderr)
+			log.Debugf("$ " + cmd.PrintableCommandArgs())
+			if err := cmd.Run(); err != nil {
+				log.Debugf("failed to copy the generated app from (%s) to the Deploy dir", path)
+				return err
 			}
 
-			buildDir, err := buildSettings.String("TARGET_BUILD_DIR")
-			if err != nil {
-				return nil, fmt.Errorf("failed to get build target dir for target (%s), error: %s", target.Name, err)
-			}
-
-			log.Debugf("Target (%s) TARGET_BUILD_DIR: %s", target.Name, buildDir)
-
-			// Split the target's TARGET_BUILD_DIR by the BUILD dir. This path will be joined to the `schemeBuildDir`
-			//
-			// xcodebuild -showBuildSettings will produce different outputs if you call it with a -workspace & -scheme or if you call it with a -project & -target.
-			// We need to call xcodebuild -showBuildSettings for all of the project targets to find the build artifacts (iOS, watchOS etc...)
-			splitTargetDir := strings.Split(buildDir, "Build/")
-			if len(splitTargetDir) != 2 {
-				log.Debugf("failed to parse build target dir (%s) for target: %s. Using the original build dir (%s)\n", buildDir, target.Name, buildDir)
-				targetDir = buildDir
-			} else {
-				targetDir = splitTargetDir[1]
-			}
+			exportedArtifacts = append(exportedArtifacts, destination)
```

Comment:
TODO: return (without error) after exporting app, to prevent recursing into app dir. For example iOS app could contain wathOS app. This was the previous behaviour so it would require version upgrade. However I do not think this is very often happesn.